File: BankManager/BankManager/Lop/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BankManager.Lop
{
    public class Account
    {
        private static int dem = 1;
        protected string soTK;
        protected double soTien;
        protected string loaiTK;
        protected DateTime ngayTao;
        protected KhachHang khachHang;
        private string password;


        public Account(string loai, KhachHang kh, double tien) {
            this.loaiTK = loai;
            this.khachHang = kh;
            this.soTien = tien;

            DateTime now = DateTime.Now;
            soTK = string.Format("{0:D2}{1:D2}{2:D4}{3:D2}", now.Day, now.Month, now.Year, dem++);
            ngayTao = DateTime.Today;

            this.password = TaoMatKhauNgauNhien(6);
        }

        public static string TaoMatKhauNgauNhien(int length)
        {
            string digits = "0123456789";
            StringBuilder password = new StringBuilder();

            Random random = new Random();
            for (int i = 0; i < length; i++)
            {
                int index = random.Next(digits.Length);
                char digit = digits[index];
                password.Append(digit);
            }

            return password.ToString();
        }


        //Getter Setter
        public string SoTK {
            get { return this.soTK; }
            set { this.soTK = value; }
        }
        public double SoTien
        {
            get { return this.soTien; }
            set { this.soTien = value; }
        }

        public string LoaiTK
        {
            get { return this.loaiTK; }
            set { this.loaiTK = value; }
        }

        public DateTime NgayTao
        {
            get { return this.ngayTao; }
            set { this.ngayTao = value; }
        }

        public KhachHang KhachHang
        {
            get { return this.khachHang; }
            set { this.khachHang = value; }
        }

        public string Password
        {
            get { return this.password; }
            set { this.password = value; }
        }
    }
}

File: BankManager_Java/DeTai7/src/main/java/com/nhk/detai7/Account.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.nhk.detai7;

import java.time.LocalDate;
import java.util.Calendar;
import java.util.GregorianCalendar;

import com.nhk.bosung.Config;

/**
 *
 * @author 84355
 */
public abstract class Account {

    private static int dem = 1;
    protected String soTK;
    protected double soTien;
    protected String loaiTK;
    protected LocalDate ngayTao;
    protected KhachHang user;

    {
        GregorianCalendar g = new GregorianCalendar();
        this.setSoTK(String.format("%02d%02d%d%04d", g.get(Calendar.DAY_OF_MONTH), g.get(Calendar.MONTH) + 1,
                g.get(Calendar.YEAR), dem++));

        this.ngayTao = LocalDate.now();
    }

    public Account(String loai, KhachHang user, double tien) {
        this.loaiTK = loai;
        this.user = user;
        this.soTien = tien;
    }

    @Override
    public String toString() {
        String tien = Config.format.format(this.soTien);
        return String.format("\n* * * * Thong tin Tai Khoan * * * *\n" +
                "*  %s\n" +
                "*  STK: %s\n" +
                "*  So du: %s\n" +
                "*  Khach hang: %s\n" +
                "*  So dien thoai: %s\n" +
                "*  So CCCD: %s\n" +
                "*  Ngay sinh: %s\n" +
                "* * * * * * * * * * * * * * * * *", this.getLoaiTK(), this.soTK, tien, this.user.getHoTen(), this.user.getSdt(), this.user.getSoCCCD(), this.user.getNgaySinh());
    }

    public void rutTien(double tien) {
        if (isDaoHan() && tien <= this.soTien) {
            this.soTien -= tien;
        } else {
            System.out.println("So tien trong tai khoan khong du!!");
        }
    }

    public void nopTien(double tien) {
        if (isDaoHan()) {
            this.soTien += tien;
        }
    }

    public abstract double tinhLai();

    protected abstract boolean isDaoHan();

    //======Getter, Setter==========
    /**
     * @return the soTK
     */
    public String getSoTK() {
        return this.soTK;
    }

    /**
     * @param soTK the soTK to set
     */
    public void setSoTK(String soTK) {
        this.soTK = soTK;
    }

    /**
     * @return the soTien
     */
    public double getSoTien() {
        return soTien;
    }

    /**
     * @param soTien the soTien to set
     */
    public void setSoTien(double soTien) {
        this.soTien = soTien;
    }

    /**
     * @return the ngayTao
     */
    public LocalDate getNgayTao() {
        return ngayTao;
    }

    /**
     * @param ngayTao the ngayTao to set
     */
    public void setNgayTao(LocalDate ngayTao) {
        this.ngayTao = ngayTao;
    }

    /**
     * @return the user
     */
    public KhachHang getUser() {
        return user;
    }

    /**
     * @param user the user to set
     */
    public void setUser(KhachHang user) {
        this.user = user;
    }

    /**
     * @return the loaiTK
     */
    public String getLoaiTK() {
        return loaiTK;
    }

    /**
     * @param loaiTK the loaiTK to set
     */
    public void setLoaiTK(String loaiTK) {
        this.loaiTK = loaiTK;
    }
    
}
